modules/aws/src/smithy4s/query/AwsQueryCodecAPI.scala

```diff
@@ -16,47 +16,91 @@
 
 package smithy4s.aws.query
 
+import cats.effect.SyncIO
+import cats.syntax.either._
 import smithy4s.http.{BodyPartial, CodecAPI, HttpMediaType, PayloadError}
 import smithy4s.schema.CompilationCache
 import smithy4s.{PayloadPath, Schema}
+import smithy4s.xml.internals.XmlDecoder
+import smithy4s.xml.internals.XmlCursor
+import smithy4s.xml.internals.XmlDecoderSchemaVisitor
+import smithy4s.internals.InputOutput
+import smithy4s.xml.XmlDocument
+import fs2.Stream
+import fs2.data.xml._
+import fs2.data.xml.dom._
 
 import java.nio.ByteBuffer
 
 private[aws] class AwsQueryCodecAPI() extends CodecAPI {
-  private val decodingError = Left(
-    PayloadError(
-      PayloadPath.root,
-      "decoding is unsupported, encoding only",
-      "TODO: integrate with XML decoders"
-    )
-  )
-
-  override type Codec[A] = AwsQueryCodec[A]
-  override type Cache = CompilationCache[AwsQueryCodec]
 
-  override def createCache(): Cache = CompilationCache.make[Codec]
+  override type Codec[A] = Either[AwsQueryCodec[A], XmlDecoder[A]]
+  override type Cache = CompilationCache[AwsQueryCodec]
 
-  override def mediaType[A](codec: Codec[A]): HttpMediaType =
-    HttpMediaType("application/x-www-form-urlencoded")
+  override def createCache(): Cache = CompilationCache.make[AwsQueryCodec]
 
   override def compileCodec[A](
       schema: Schema[A],
       cache: Cache
-  ): Codec[A] = schema.compile(new AwsSchemaVisitorAwsQueryCodec(cache))
+  ): Codec[A] =
+    schema.hints match {
+      case InputOutput.hint(InputOutput.Input) =>
+        Left(schema.compile(new AwsSchemaVisitorAwsQueryCodec(cache)))
```

Comment:
I'll have to use the `InputOutput.Input(serviceId, version)` somehow.

Reply:
yeah. You'll have to modify the QueryCodec to add the static information you get from the hint you created.